nlp_utilities_test.jl

```diff
@@ -0,0 +1,254 @@
+using JuMP
+using Ipopt
+using Test
+using FiniteDiff
+
+include("nlp_utilities.jl")
+
+#=
+# Test JuMP Hessian and Jacobian
+
+From JuMP Tutorial for Querying Hessians:
+https://github.com/jump-dev/JuMP.jl/blob/301d46e81cb66c74c6e22cd89fb89ced740f157b/docs/src/tutorials/nonlinear/querying_hessians.jl#L67-L72
+=#
+function create_nonlinear_jump_model()
```

Comment:
I am currently testing the sensitivities using finite differences and getting a few of them wrong 

```
julia> test_compute_derivatives_Finite_Diff()
┌ Warning: Relaxation needed
└ @ Main ~/Workspace/DiffOpt.jl/nlp_utilities.jl:343
QP_JuMP: Min (x[1]² - 2 x[1] + 1) + (p3 * ((-x[1]² + x[2]) ^ 2.0))
Subject to
 g_2 : (p * (x[1]² + 2 x[2]*x[1] + x[2]²)) - p2 ≤ 0
 g_1 : x[1]² - p ≤ 0
 p ∈ MathOptInterface.Parameter{Float64}(1.0)
 p2 ∈ MathOptInterface.Parameter{Float64}(2.0)
 p3 ∈ MathOptInterface.Parameter{Float64}(100.00000149011612)

Some sensitivities are not correct: 

Primal var: x[1] | Δs: 0.23807735138884525 | Δs_fd: -0.0002735186815261841
Primal var: x[2] | Δs: -0.23878427590905257 | Δs_fd: -0.00043358781188726427
Slack var: g_1 : x[1]² - p ≤ 0 | Δs: 0.37533303814885505 | Δs_fd: 0.0
Dual var: g_2 : (p * (x[1]² + 2 x[2]*x[1] + x[2]²)) - p2 ≤ 0 | Δs: 43.49539673020088 | Δs_fd: 5.8398510795086626e-5
Dual var: g_1 : x[1]² - p ≤ 0 | Δs: -201.20123731450576 | Δs_fd: -3.067666263234514e-10
--------------------
```